joomla/joomla-cms
# Cleanup and optimization in FinderIndexerDrivers

### Summary of Changes
- Pulled methods, of common functionality in at least two drivers, up to abstract class. These are  overridden in inheriting drivers to provide a different functionality, where needed.
- Save a few cycles by introducing a query template for `addTokensToDb()` to **clone** instead of constantly instantiating and creating columns, which would result in unnecessary expensive calls to `query` construction and `quoteName`
- Save a few cycles by instantiating and using a `db` property. No need to constantly calling `JFactory->getDbo()` to determine if a database is created.

### Testing Instructions
No change of behavior is expected.
- Code Review (best using the `w=1` parameter in the **[url](https://github.com/joomla/joomla-cms/pull/13511/files?w=1))**, since also whitespaces were corrected
- Test functionality of indexer. It would be good if someone could also test with PostgeSQL and MSSQL besides MySQL.

### Documentation Changes Required
None
